KarhooUISDKUITests/Checkout/CheckoutSnapshotSpec.swift

```diff
@@ -0,0 +1,248 @@
+//
+//  CheckoutSnapshotSpec.swift
+//  KarhooUISDKUITests
+//
+//  Created by Bartlomiej Sopala on 02/02/2023.
+//  Copyright © 2023 Flit Technologies Ltd. All rights reserved.
+//
+
+import Quick
+import Nimble
+import SnapshotTesting
+import KarhooSDK
+import Combine
+import KarhooUISDKTestUtils
+@testable import KarhooUISDK
+
+class CheckoutSnapshotSpec: QuickSpec {
+    
+    override func spec() {
+        describe("Checkout") {
+            var navigationController: NavigationController!
+            var sut: KarhooCheckoutViewController!
+            var viewModel: KarhooCheckoutViewModel!
+            var quote: Quote!
+            
+            beforeEach {
+                KarhooUI.set(configuration: KarhooTestConfiguration())
+                let mockVC = MockViewController().then {
+                    $0.loadViewIfNeeded()
+                }
+                navigationController = NavigationController(rootViewController: mockVC, style: .primary)
+                sut = KarhooCheckoutViewController()
+            }
+            
+            context("when Checkout is opened without poi and with scheduled ride") {
+                beforeEach {
+                    quote = TestUtil.getRandomQuote(
+                        fleetName: "Fleet name",
+                        categoryName: "Category name",
+                        type: "type"
+                    )
+                    let journeyDetails = JourneyDetails.mockWithScheduledDate()
+                    viewModel = KarhooCheckoutViewModel(
+                        quote: quote,
+                        journeyDetails: journeyDetails,
+                        bookingMetadata: nil,
+                        router: MockCheckoutRouter()
+                    )
+                    viewModel.legalNoticeViewModel.shouldShowView = false
+                    sut.setupBinding(viewModel)
+                    navigationController.pushViewController(sut, animated: false)
```

Comment:
Seems like this naviation methods should be moved to main beforeEach (lines 27-32).